### PR TITLE
Added python bindings for tensor descriptor

### PR DESCRIPTION
**Pull request by amkarn258**
### Details:
 Bindings of methods get_lower_value/set, get_upper_value/set and get_value_symbol/set added for tensor descriptor along with their respective tests.

### Tickets:
 - *140495*

Closes Issue [#24909](https://github.com/openvinotoolkit/openvino/issues/24909)

### CODE REVIEW

**Comment by p-wysocki** [comment]
build_jenkins

File: src/bindings/python/tests/test_runtime/test_input_node.py [hunk 1]
```diff
@@ -133,3 +133,35 @@ def test_input_update_rt_info(device):
...
+    core = Core()
+    model = get_relu_model()
+    compiled_model = core.compile_model(model, device)
+    tensor = compiled_model.output(0).get_tensor()
+    partial_shape = tensor.get_partial_shape().to_shape()
+    tensor_size = np.prod(partial_shape)
+    values = [Symbol() for _ in range(tensor_size)]
+    a = tensor.get_value_symbol()
+    tensor.set_value_symbol(values)
+    assert tensor.get_value_symbol() == values
```
Comment (p-wysocki) [inline src/bindings/python/tests/test_runtime/test_input_node.py:167]:
```suggestion
    assert tensor.get_value_symbol() == values

```

**Review by p-wysocki — APPROVED** [review]
Thank you for your contribution, looks good to me! @akuporos do you think we need any documentation updates as a follow up?

File: src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp [hunk 2]
```diff
@@ -106,6 +106,63 @@ void regclass_graph_descriptor_Tensor(py::module m) {
...
                 :rtype: string
              )");
+   
+   tensor.def("get_lower_value",
+               &ov::descriptor::Tensor::get_lower_value,
+               R"(
+                Returns the lower bound of the tensor.
+
+                :return: Lower bound.
+                :rtype: Tensor
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp:116]:
```suggestion
                :rtype: openvino.runtime.Tensor
```
Better to have it like this for proper documentation generation

File: src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp [hunk 3]
```diff
@@ -106,6 +106,63 @@ void regclass_graph_descriptor_Tensor(py::module m) {
...
+                :type lower_bound: Tensor
+             )");
+
+    tensor.def("get_upper_value",
+               &ov::descriptor::Tensor::get_upper_value,
+               R"(
+                Returns the upper bound of the tensor.
+
+                :return: Upper bound.
+                :rtype: Tensor
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp:135]:
```suggestion
                :rtype: openvino.runtime.Tensor
```

File: src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp [hunk 4]
```diff
@@ -106,6 +106,63 @@ void regclass_graph_descriptor_Tensor(py::module m) {
...
+             )");
+
+    tensor.def("set_upper_value",
+               &ov::descriptor::Tensor::set_upper_value,
+               py::arg("upper_bound"),
+               R"(
+                Sets the upper bound of the tensor.
+
+                :param upper_bound: Sets the upper bound value.
+                :type upper_bound: Tensor
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp:145]:
```suggestion
                :type upper_bound: openvino.runtime.Tensor
```

File: src/bindings/python/tests/test_runtime/test_input_node.py (hunk 1 above)
Comment (akuporos) [inline src/bindings/python/tests/test_runtime/test_input_node.py:167]:
```suggestion
    assert tensor.get_value_symbol() == values

```
to fix code style

File: src/bindings/python/tests/test_runtime/test_input_node.py [hunk 5]
```diff
@@ -133,3 +133,35 @@ def test_input_update_rt_info(device):
...
+    lower_value_tensor = Tensor(lower_value.reshape(partial_shape))
+    upper_value = np.ones(tensor_size, dtype=np.float32)
+    upper_value_tensor = Tensor(upper_value.reshape(partial_shape))
+    
+    tensor.set_lower_value(lower_value_tensor)
+    retrieved_lower_value = tensor.get_lower_value().data
+    tensor.set_upper_value(upper_value_tensor)
+    retrieved_upper_value = tensor.get_upper_value().data
+    assert np.array_equal(retrieved_lower_value, lower_value_tensor.data) and np.array_equal(retrieved_upper_value, upper_value_tensor.data)
+
```
Comment (akuporos) [inline src/bindings/python/tests/test_runtime/test_input_node.py:156]:
```suggestion


```
to fix code style

File: src/bindings/python/tests/test_runtime/test_output_node.py [hunk 6]
```diff
@@ -31,3 +31,19 @@ def test_output_names():
     more_names = {"yet_another_name", "input1"}
     param.output(0).add_names(more_names)
     assert param.output(0).get_names() == names.union(more_names)
+
```
Comment (akuporos) [inline src/bindings/python/tests/test_runtime/test_output_node.py:34]:
```suggestion


```
to fix code style

File: src/bindings/python/tests/test_runtime/test_output_node.py [hunk 7]
```diff
@@ -31,3 +31,19 @@ def test_output_names():
...
+def test_tensor_bounds():
+    param = ops.parameter([1, 64], Type.f32)
+    tensor = param.output(0).get_tensor()
+    values = Tensor(np.zeros([1, 64], dtype=np.float32))
+    tensor.set_lower_value(values)
+    tensor.set_upper_value(values)
+    assert np.array_equal(tensor.get_lower_value().data, values.data) and np.array_equal(tensor.get_upper_value().data, values.data)
+
+def test_output_symbol():
+    param = ops.parameter([1,64], Type.f32)
```
Comment (akuporos) [inline src/bindings/python/tests/test_runtime/test_output_node.py:44]:
```suggestion
    param = ops.parameter([1, 64], Type.f32)
```
to fix code style job

File: src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp [hunk 8]
```diff
@@ -106,6 +106,63 @@ void regclass_graph_descriptor_Tensor(py::module m) {
...
+             )");
+
+    tensor.def("get_value_symbol",
+               &ov::descriptor::Tensor::get_value_symbol,
+               R"(
+                Returns the List of symbols.
+
+                :return: List of Symbols.
+                :rtype: List
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/graph/descriptors/tensor.cpp:154]:
```suggestion
                :rtype: List[openvino.runtime.Symbol]
```

**Comment by akuporos** [comment]
Hi @amkarn258,

Thank you for your contribution!

In order to fix C++ code style, please run next commands:
If you don't have installed clang-format-15, so let's install it:
`sudo apt --assume-yes install clang-format-15`
After that run cmake with -DENABLE_PROFILING_ITT=ON -DSELECTIVE_BUILD=COLLECT in order to enable codestyle check for ITT collector:
`cmake -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_PROFILING_ITT=ON -DSELECTIVE_BUILD=COLLECT -B build`
And create code style diff:
`cmake --build build --target clang_format_fix_all -j8`

File: thirdparty/itt_collector/sea_itt_lib/IttNotifyStdSrc.cpp [hunk 9]
```diff
@@ -1577,15 +1577,13 @@ void FillApiList(__itt_api_info* api_list_ptr) {
 #define ITT_STUB_IMPL_ORIG(name) ITT_STUB_IMPL(name)
 #ifdef _DEBUG  // dangerous stub that doesn't return anything (even when expected) but records the function call for
                // statistics sake
-#    define ITT_STUB_NO_IMPL(fn)                                              \
```
Comment (akuporos) [inline thirdparty/itt_collector/sea_itt_lib/IttNotifyStdSrc.cpp:1580]: Could you please undo the changes in this file?
They were introduced together with the fix of c++ code style

Comment (amkarn258) [reply thirdparty/itt_collector/sea_itt_lib/IttNotifyStdSrc.cpp]: Done

File: src/bindings/python/tests/test_runtime/test_input_node.py (hunk 1 above)
Comment (amkarn258) [reply src/bindings/python/tests/test_runtime/test_input_node.py]: Hi, 

Have fixed the python linter using the steps in this [documentation](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/test_examples.md#check-the-codestyle-of-python-api)

**Comment by amkarn258** [comment]
Hi, 

Have added the fixes for cpp codestyle and python linter.

**Review by akuporos — APPROVED** [review]
Good work!

**Comment by akuporos** [comment]
build_jenkins
